File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==1)return nums[0];
       int  maxsum=INT_MIN;

      int  currsum=0;
        for(int i=0;i<nums.size();i++)
            {
            currsum+=nums[i];
            
            
            if(maxsum<currsum)
            
                {maxsum=currsum;}
            
            if(currsum<0)

                {currsum=0;}
            }
        return maxsum;
            
        
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=0,mx=Integer.MIN_VALUE;
        
        if(nums.length==1)return nums[0];
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            
            mx=Math.max(mx,sum);
            
            if(sum<0)sum=0;
            
            
            
            
        }
        return mx;
        
        
        
        
        
        
        
    }
}